src/DotNetProjectFile.Analyzers/MsBuild/Node.cs

```diff
@@ -5,8 +5,18 @@
 
 namespace DotNetProjectFile.MsBuild;
 
+public abstract class Node<T> : Node
+{
+    protected Node(XElement element, Node? parent, Project? project)
+        : base(element, parent, project) { }
+
+    public sealed override object? Val => Value;
+
+    public abstract T? Value { get; }
```

Comment:
Perhaps turn this into `public virtual T? Value => Convert<T?>(Element.Value);` To reduce the amount of copy-paste code happening in nodes that don't really have much behaviour.

Reply:
I like that suggestion.